src/frontend/src/flows/manage/migration.test.ts

```diff
@@ -0,0 +1,88 @@
+import { domainWarning } from "../manage";
+import { DeviceData } from "../../../generated/internet_identity_types";
+
+function onOrigin(origin: string, fn: () => void) {
+  const oldOrigin = window.origin;
+  Object.defineProperty(window, "origin", {
+    writable: true,
+    value: origin,
+  });
+
+  fn();
+  Object.defineProperty(window, "origin", {
+    writable: true,
+    value: oldOrigin,
+  });
+}
+
+const recoveryPhrase: DeviceData = {
+  alias: "Recovery Phrase",
+  origin: [],
+  protection: { unprotected: null },
+  pubkey: undefined as any,
+  key_type: { seed_phrase: null },
+  purpose: { recovery: null },
+  credential_id: [],
+};
+
+const authenticator: DeviceData = {
+  alias: "My Authenticator",
+  origin: [],
+  protection: { unprotected: null },
+  pubkey: undefined as any,
+  key_type: { unknown: null },
+  purpose: { authentication: null },
+  credential_id: [],
+};
+
+test("recovery phrases don't have origin warnings", () => {
+  onOrigin("https://identity.ic0.app", () => {
+    expect(domainWarning(recoveryPhrase)).toBe(undefined);
+    expect(
+      domainWarning({ ...recoveryPhrase, origin: ["https://elsewhere"] })
+    ).toBe(undefined);
+  });
+
+  onOrigin("https://identity.internetcomputer.org", () => {
+    expect(domainWarning(recoveryPhrase)).toBe(undefined);
+    expect(
+      domainWarning({ ...recoveryPhrase, origin: ["https://elsewhere"] })
+    ).toBe(undefined);
+  });
+});
+
+test("no origin on legacy domain is not warning", () => {
+  onOrigin("https://identity.ic0.app", () => {
+    expect(domainWarning({ ...authenticator, origin: [] })).toBe(undefined);
+  });
+});
+
+test("bad origin on legacy domain is warning", () => {
```

Comment:
I noticed that on `LEGACY_II_URL` devices added on `OFFICIAL_II_URL` get treated the same way as any other 'bad' URL.
For me this feels counter-intuitive. I.e. we specifically told people to do that in order to be prepared if `ic0.app` gets taken down. To then warn with a red exclamation mark is ...unexpected?
I think it is fine to inform the user, but I would suggest something that does not imply an action is necessary. I.e. a blue ℹ️ with "This device is bound to the new II domain and cannot be used on `identity.ic0.app`."

Reply:
Sure! Happy to discuss this offline, I don't have strong feelings about this. Resolving this since it's not directly relevant to the tests, but thanks for bringing it up!